src/ansys/fluent/core/systemcoupling.py

```diff
@@ -73,19 +74,25 @@ def solve(self) -> None:
     def __get_syc_setup(self) -> dict:
         def get_scp_string() -> str:
             """Get SCP file contents in the form of the XML string."""
+
             fluent_cwd = self._solver.connection_properties.cortex_pwd
             scp_file_name = os.path.join(fluent_cwd, "fluent.scp")
             self._solver.setup.models.system_coupling.write_scp_file(
                 file_name=scp_file_name
             )
+            local_scp_file_name = scp_file_name
+            if self._solver.connection_properties.inside_container:
```

Comment:
Would `<session>.fluent_connection._remote_instance` be a better way to check if remote?

Alternatively, I think we don't even need to check if remote here, in my understanding `<session>.download()` won't do anything nor raise errors if PyPIM isn't configured and if it isn't running remote. @hpohekar is this correct?

Otherwise we'd probably need a more elegant way to check if running remote

Reply:
I think this is correct. I have been looking into the Fluent implementation just recently and my understanding is that `download` reduces to a no-op if PyPIM isn't configured.

Reply:
@raph-luc No.

We raise `PyPIMConfigurationError` if PyPIM is not configured.

![image](https://github.com/ansys/pyfluent/assets/106588300/3ea143c8-189d-4a0b-b656-b6f56e425b3d)

![image](https://github.com/ansys/pyfluent/assets/106588300/e5fdb4d9-e58c-4a56-b883-93c54bf8b8e7)

Reply:
Ok thanks for the clarification @hpohekar, then in my understanding the description of `PimFileTransferService` is incorrect as it implies these should be no-ops:
https://github.com/ansys/pyfluent/blob/caddd75a20c8f72975295b4f3d297fd2a2588962/src/ansys/fluent/core/utils/file_transfer_service.py#L15-L19

cc @seanpearsonuk 

@ochernuk @iboyd-ansys in this case checking `<session>.fluent_connection._remote_instance` is probably the way to go for now

Reply:
My apologies - it turns out I was looking at an older version of the PyFluent code. It looks to me that it used to behave more like the way @raph-luc described.

Reply:
@raph-luc The intention has always been that `upload()` and `download()` are no-ops if no service is available. That's so that the same script works both in local and remote contexts. This was certainly the status in the past, and there ought to have been a test for that (I believe that there was). @hpohekar Has there been a conscious change of behaviour in this respect?

Reply:
@seanpearsonuk @raph-luc 

Earlier we had only single file transfer service (PimFileTransferService). Now we can plug in any file transfer service which has upload/download service therefore we are raising appropriate exception if particular service is not configured.

Reply:
@hpohekar OK, I see. Please could you make a simple change so that upload an dowload calls can just do nothing and not raise. That is still valid. It still comes back to the script reuse requirement.

Reply:
@seanpearsonuk Yes, sure.

Reply:
Thanks